python/cuspatial/cuspatial/core/binpreds/binpreds.py

```diff
@@ -129,6 +134,10 @@ def __call__(self) -> cudf.Series:
 
 
 class ContainsProperlyBinpred(BinaryPredicate):
+    def __init__(self, lhs, rhs, align=True, mode="pairs"):
```

Comment:
Making sure @harrism 's question is not falling out of sync:

> Looking at geopandas APIs, I see a lot of named mode parameters rather than boolean. how is commonly used, e.g.
> 
> lanes_districts = gpd.sjoin(districts, bike_lane, how='inner', predicate='intersects')
> lanes_districts
> 
> So should we use how="allpairs" (and how="pairwise") instead of a Boolean parameter? Perhaps how might be needed also for other things, so maybe we need a new name for this, e.g. parallelism="allpairs" with options of "all_pairs", "pairwise", "one_to_all", etc.
> 
> Thoughts?

https://github.com/rapidsai/cuspatial/pull/910#discussion_r1113812489

Reply:
Thanks @isVoid